functorch/experimental/_while_loop.py

```diff
@@ -81,6 +82,44 @@ def trace_while_loop(proxy_mode, func_overload, cond_fun, body_fun, init_val):
     out = pytree.tree_map(lambda fake_t: torch.empty_like(fake_t), init_val)
     return track_tensor_tree(out, out_proxy, constant=None, tracer=proxy_mode.tracer)
 
+@while_loop.py_impl(torch._C._functorch.TransformType.Functionalize)
+def while_loop_functionalize(interpreter, cond_fun, body_fun, init_val):
+    """
+    Functionalization implementation for torch.while_loop. Currently:
+      1. We don't allow any input mutation inside the map function
+      2. Our check for above condition is not exhaustive
+    """
+    reapply_views = interpreter.functionalize_add_back_views()
+    mode = 'mutations' if reapply_views else 'mutations_and_views'
+    # At this point, we will see functionalized tensors, so need to unwrap them first
+    unwrapped_args = _unwrap_all_tensors_from_functional(init_val, reapply_views=reapply_views)
+
+    functional_cond_fun = functionalize(cond_fun, remove=mode)
+    functional_body_fun = functionalize(body_fun, remove=mode)
+
+    with interpreter.lower():
+        if _has_potential_branch_input_mutation(functional_cond_fun, unwrapped_args):
+            raise UnsupportedAliasMutationException(
+                "torch.ops.while_loop cond_fun mutates the input!"
+            )
+        if _has_potential_branch_input_alias(functional_cond_fun, unwrapped_args):
+            raise UnsupportedAliasMutationException(
+                "torch.ops.while_loop cond_fun is mutates the input!"
```

Comment:
This should probably be `torch.ops.while_loop cond_fun returns an alias of the input!` instead of "mutates"? (it looks flipped with the error message 5 lines down).